test/Table_Tests/src/Common_Table_Spec.enso

```diff
@@ -0,0 +1,143 @@
+from Standard.Base import all
+import Standard.Test
+import Standard.Test.Problems
+
+import Standard.Base.Error.Problem_Behavior
+import Standard.Base.Error.Warnings
+import Standard.Table.Data.Matching
+from Standard.Table.Error as Table_Errors import all
+from Standard.Table.Data.Column_Selector as Column_Selector_Module import all
+
+## A common test suite for shared operations on the Table API.
+
+   It is meant to be able to be re-used by tests for the in-memory and various
+   database backends.
+
+   Arguments:
+   - prefix: A prefix to attach to the names of test groups, for easier
+     identification.
+   - table_builder: A function which takes a list of column descriptions and
+     builds a Table using the backend that is meant to be tested. Each column
+     description is a triple of column name, column type and a vector containing
+     column elements.
+
+   TODO [RW] the Any in return type of the builder should ideally be replaced with the Table interface, once that is supported.
+spec : Text -> (Vector -> Any) -> Boolean -> Nothing
+spec prefix table_builder supports_case_sensitive_columns =
+    Test.group prefix+"Select" <|
+        table =
+            col1 = ["foo", Integer, [1,2,3]]
+            col2 = ["bar", Integer, [4,5,6]]
+            col3 = ["Baz", Integer, [7,8,9]]
+            col4 = ["foo_1", Integer, [10,11,12]]
+            col5 = ["foo_2", Integer, [13,14,15]]
```

Comment:
Might be worth including a Regex looking name as part of the suite
- `ab.123` for example